### ISSUE TITLE
Please, recompile Certes to use newest BouncyCastle and NewtonJson

### ISSUE THREAD

**Issue by hhblaze** [fszlin/certes#177]
Hi, thank you for the product!!!

One of my certificates was not updated automatically, I have recompiled Certes to use BouncyCastle 1.8.5 and latest NewtonJson - problem was fixed.

**Comment by webprofusion-chrisc** [comment]
Hi, do you have any logs or errors for the failed renewal? Otherwise it could just be a coincidence that recompiling Certes fixed the issue.

**Comment by hhblaze** [comment]
There is one log. I don't think it was a coincidence. Program was restarted and run several times, no code was changed, then recompilation of Certes to work with new BountyCastle + new NewtonJson. And it has updated the failed domain.

Here is log, before update:

Certes.AcmeRequestException: Fail to load resource from 'https://acme-v02.api.letsencrypt.org/acme/finalize/41207772/313637087'. urn:ietf:params:acme:error:malformed: Order's status ("invalid") is not acceptable for finalization at Certes.Acme.IAcmeHttpClientExtensions.d__0`1.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Certes.Acme.OrderContext.d__2.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Certes.IOrderContextExtensions.d__0.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Certes.IOrderContextExtensions.d__2.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at GM_ACME.ACME.AcmeHandler.d__12.MoveNext()

**Comment by webprofusion-chrisc** [comment]
Thanks, are you doing DNS or http validation? Your certificate order was invalid which means the challenge failed (either http or DNS). 

I'm sure Certes will be updated to the latest versions of the dependencies, however there's nothing to prove that's why you were successful in this instance.

**Comment by hhblaze** [comment]
It was DNS validation.

**Comment by webprofusion-chrisc** [comment]
If you are completing a DNS challenge for both *.domain.com and domain.com they both will be required to add a TXT record value for _acme-challenge.domain.com - with some providers you can have multiple values, others you have to do one at a time. So you can have 2 challenges where 1 fails, then when you try it again the other challenge is already valid and you only have one left to do, so the remaining challenge succeeds and the overal certificate request completes. 

So basically it's quite common for DNS validation to take 2 attempts unless you are only requesting one name.

**Comment by hhblaze** [comment]
Interesting, it was made only for *.domain.com. For domain.com challenge must be initiated in 2 days due to the info I see in RenewalScheduler. 
So, there is a chance to take a look on this behavior also.
